src/Features/Core/Portable/AddConstructorParametersFromMembers/AddConstructorParametersFromMembersCodeRefactoringProvider.cs

```diff
@@ -151,5 +154,37 @@ static CodeAction GetOptionalContructorParametersCodeAction(Document document, C
                     document, constructorCandidate, containingType, missingOptionalParameters, useSubMenuName);
             }
         }
+
+        public async Task<ImmutableArray<IntentProcessorResult>> ComputeIntentAsync(Document priorDocument, TextSpan priorSelection, Document currentDocument, string serializedIntentData, CancellationToken cancellationToken)
+        {
+            var actions = await AddConstructorParametersFromMembersAsync(priorDocument, priorSelection, cancellationToken).ConfigureAwait(false);
+            if (actions.IsEmpty)
+            {
+                return ImmutableArray<IntentProcessorResult>.Empty;
+            }
+
+            // This provider generates nested actions, so we need to flatten them before we compute the edits.
+            actions = actions.SelectMany(a => GetNestedActions(a)).ToImmutableArray();
+
+            using var _ = ArrayBuilder<IntentProcessorResult>.GetInstance(out var results);
+
+            foreach (var action in actions)
+            {
+                var addParamAction = (AddConstructorParametersCodeAction)action;
+
+                // This provider only returns CodeActions with a single ApplyChangesOperation
+                var operations = await addParamAction.GetOperationsAsync(cancellationToken).ConfigureAwait(false);
+                var applyChangesOperation = operations.Single() as ApplyChangesOperation;
+                Contract.ThrowIfNull(applyChangesOperation);
```

Comment:
might as well just blindly cast to ApplyChangesOperation.

I don't love this.  but i have no better solution.  esp. as i assume you want to go through the full codepath so that simplication/formatting/etc. is involved.  will have to noodle on if ther's a nicer way to accomplish this that isn't so hacky.

Reply:
I found a nice helper API on code actions that I've switched to - hopefully looks better